datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -739,4 +824,76 @@ mod tests {
         assert_eq!(ints.value(2), 75);
         assert_eq!(ints.value(3), 16);
     }
+
+    #[test]
+    fn test_truncate_32() {
```

Comment:
I think this test would be better written using sqllogictest (I have left comment about how to do this)